src/low-level/imap/acl.c

```diff
@@ -347,6 +347,7 @@ int mailimap_acl_myrights(mailimap * session,
           * result = (struct mailimap_acl_myrights_data *)ext_data->ext_data;
```

Comment:
Add `ext_data->ext_data = NULL` here to transfer ownership.

Reply:
Thanks!  I've modified and tested it. Please see this merge request again.

I also modified mailimap_extension_data_free to handle the case of `data->ext_data == NULL`